src/thermostats/TemperatureControl.h

```diff
@@ -116,7 +116,8 @@ class ControlRegionT : public CuboidRegionObs
 	unsigned int _nNumSlabs;
 	double _dSlabWidth;
 
-	std::vector<LocalAndGlobalThermostatVariables> _thermVars;
+	std::vector<std::vector<LocalThermostatVariables>> _localThermVarsThreads;
+	std::vector<GlobalThermostatVariables> _globalThermVars;
```

Comment:
is the class `LocalAndGlobalThermostatVariables` now unused? Maybe delete it as it looks sketchy to begin with...

Also: Comment these new variables especially the vec<vec<>> buffers
```suggestion
	std::vector<std::vector<LocalThermostatVariables>> _localThermVarsThreadBuffer;
	std::vector<GlobalThermostatVariables> _globalThermVars;
```